backend_new/src/utilities/build-filter.ts

```diff
@@ -0,0 +1,99 @@
+import { HttpException, HttpStatus } from '@nestjs/common';
+import { Prisma, UserAccounts, UserRoles } from '@prisma/client';
+import { Compare } from '../dtos/shared/base-filter.dto';
+import { UserFilterKeys } from '../enums/user_accounts/filter-key-enum';
+
+type filter = {
+  $comparison: Compare;
+  $include_nulls: boolean;
+  value: any;
+  key: string;
+};
+
+/*
+  This constructs the "where" part of a prisma query
+  Because the where clause is specific to each model we are working with this has to be very generic.
+  It only constructs the actual body of the where statement, how that clause is used must be managed by the service calling this helper function
+*/
+export function buildFilter(
+  filter: filter,
+  user?: UserAccounts & { roles: UserRoles },
+): any {
+  const toReturn = [];
+  const comparison = filter['$comparison'];
+  const includeNulls = filter['$include_nulls'];
+  const filterValue = filter.value;
+
+  if (filter.key === UserFilterKeys.isPortalUser) {
+    // TODO: addIsPortalUserQuery(filter.value, user);
+  }
+
+  if (comparison === Compare.IN) {
+    toReturn.push({
+      in: String(filterValue)
+        .split(',')
+        .map((s) => s.trim().toLowerCase())
+        .filter((s) => s.length !== 0),
+      mode: Prisma.QueryMode.insensitive,
+    });
+    if (includeNulls) {
+      toReturn.push({
+        equals: null,
+      });
+    }
+  } else if (comparison === Compare['<>']) {
+    toReturn.push({
+      not: {
+        equals: filterValue,
+      },
+      mode: Prisma.QueryMode.insensitive,
+    });
+    if (includeNulls) {
+      toReturn.push({
+        equals: null,
+      });
+    }
+  } else if (comparison === Compare['=']) {
+    toReturn.push({
```

Comment:
some repetition here

Reply:
good 👀  I pulled the if (includeNulls) out, lemme know if whats left is acceptable or if I should look to simplify this further